scipy/signal/setup.py

```diff
@@ -22,8 +23,17 @@ def configuration(parent_package='', top_path=None):
 
     config.add_extension(
         '_spectral', sources=['_spectral.c'])
-    config.add_extension(
-        '_max_len_seq_inner', sources=['_max_len_seq_inner.c'])
+
+    if int(os.environ.get('SCIPY_USE_PYTHRAN', 0)):
+        import pythran
+        ext = pythran.dist.PythranExtension(
+            'scipy.spatial._max_len_seq_inner',
```

Comment:
It's possible to declare various overloads, with

```python
#pythran export(int64[:], int8[:], int or int8 or int16, int64, int8[:])
```

And the likes. I don't think it's worth the extra compilation time and would rather require a cast at call site.
Pythran is rather strict on the type argument and doesn't do much implicit casts. That's required to distinguish between otherwise similar overloads.

Reply:
> I don't think it's worth the extra compilation time and would rather require a cast at call site.

Agreed for this PR, where those parameters are scalars. If they're arrays, casts may be expensive - we'd then typically want to support at least int32/int64 and float32/float64.